### PR TITLE
Update the minion hostname in a regular basis, based on the grain information.

### PR DESCRIPTION
**Pull request by ereslibre**
Fixes: bsc#1103307

### CODE REVIEW

File: spec/controllers/internal_api/v1/pillars_controller_spec.rb [hunk 1]
```diff
@@ -310,6 +310,9 @@
       {
         registries:          [],
         system_certificates: [],
+        dex:                 {
+          connectors: []
+        },
```
Comment (flavio) [inline spec/controllers/internal_api/v1/pillars_controller_spec.rb]: I think this is unrelated with this PR, isn't it?

File: app/models/minion.rb [hunk 2]
```diff
@@ -25,7 +25,8 @@ def self.update_grains
 
         tx_update_reboot_needed = minion_grains["tx_update_reboot_needed"] || false
         tx_update_failed = minion_grains["tx_update_failed"] || false
-        minion.update_columns online:                  online,
+        minion.update_columns fqdn:                    minion_grains["fqdn"],
+                              online:                  online,
```
Comment (flavio) [inline app/models/minion.rb]: Is the reconciliation loop going to keep this grain in sync with the database?

File: spec/controllers/internal_api/v1/pillars_controller_spec.rb (hunk 1 above)
Comment (vitoravelino) [reply spec/controllers/internal_api/v1/pillars_controller_spec.rb]: It will disappear after a rebase. An old PR was merged without rebasing and introduced this small issue that was already fixed.

File: app/models/minion.rb (hunk 2 above)
Comment (MaximilianMeister) [reply app/models/minion.rb]: @flavio yes it does. it [calls exactly this method here](https://github.com/kubic-project/velum/blob/master/lib/tasks/salt.rake#L18)

File: spec/controllers/internal_api/v1/pillars_controller_spec.rb (hunk 1 above)
Comment (mjura) [reply spec/controllers/internal_api/v1/pillars_controller_spec.rb]: lol

**Comment by flavio** [comment]
It will LGTM to me after the rebase is done

File: spec/controllers/internal_api/v1/pillars_controller_spec.rb (hunk 1 above)
Comment (ereslibre) [reply spec/controllers/internal_api/v1/pillars_controller_spec.rb]: Yes, this was in purpose because travis was red because of this. Rebasing now.

**Comment by ereslibre** [comment]
Rebased.
